regolith/helpers/u_milestonehelper.py

```diff
@@ -0,0 +1,148 @@
+    subpi.add_argument("--number",
+                        help="number of the milestone to update from numbered list")
+    subpi.add_argument("--due_date",
+                       help="new due date of the milestone in ISO format (YYYY-MM-DD) "
+                            "required to add a new milestone")
+    subpi.add_argument("-s", "--status",
+                       help="initial of the status of the milestone/deliverable: "
+                            "p = proposed, s = started, f = finished, c = cancelled")
+    subpi.add_argument("--name",
+                       help="name of the new milestone. "
+                            "required to add a new milestone")
+    subpi.add_argument("-o", "--objective",
+                       help="objective of the new milestone. "
+                            "required to add a new milestone")
+    subpi.add_argument("-t", "--type",
+                       help="initial of type of the new milestone "
+                            "m = meeting, r = release, p = pull request, o = other")
+    # Do not delete --database arg
+    subpi.add_argument("--database",
+                       help="The database that will be updated.  Defaults to "
+                            "first database in the regolithrc.json file.")
+    return subpi
+
+class MilestoneUpdaterHelper(DbHelperBase):
+    """Helper for updating milestones to the projecta collection.
+    """
+    # btype must be the same as helper target in helper.py
+    btype = "u_milestone"
+    needed_dbs = [f'{TARGET_COLL}']
+
+    def construct_global_ctx(self):
+        """Constructs the global context"""
+        super().construct_global_ctx()
+        gtx = self.gtx
+        rc = self.rc
+        rc.coll = f"{TARGET_COLL}"
+        if not rc.database:
+            rc.database = rc.databases[0]["name"]
+        gtx[rc.coll] = sorted(
+            all_docs_from_collection(rc.client, rc.coll), key=_id_key
+        )
+        gtx["all_docs_from_collection"] = all_docs_from_collection
+        gtx["float"] = float
+        gtx["str"] = str
+        gtx["zip"] = zip
+
+    def db_updater(self):
+        rc = self.rc
+        key = rc.projectum_id
+        allowed_types = {'m': 'meeting', 'r':'release', 'p':'pr', 'o':'other'}
+        allowed_status = {'p':'proposed', 's':'started', 'f':'finished', 'c':'cancelled'}
+        coll = self.gtx[rc.coll]
+        pdocl = list(filter(lambda doc: doc["_id"] == key, coll))
+        if len(pdocl) == 0:
+            raise RuntimeError(
+                "This entry appears to not exist in the collection")
+        filterid = {'_id': key}
+        current = rc.client.find_one(rc.database, rc.coll, filterid)
+        milestones = current.get('milestones')
+        deliverable = current.get('deliverable')
+        kickoff = current.get('kickoff')
+        deliverable['identifier'] = 'deliverable'
+        kickoff['identifier'] = 'kickoff'
+        for item in milestones:
+            item['identifier'] = 'milestones'
```

Comment:
I added those identifiers because I need to know if it is a milestone, deliverable, or kickoff (when I am going to make changes to the doc), and I didn't know how to keep that info without doing this. I am trying to find a more efficient way to do this.